tsec-http4s/src/test/scala/tsec/authentication/TSecAuthServiceSpec.scala

```diff
@@ -0,0 +1,36 @@
+package tsec.authentication
+
+import java.util.UUID
+
+import cats.data.OptionT
+import cats.effect.IO
+import org.http4s._
+import org.http4s.dsl.io._
+import org.http4s.headers.{Authorization => H4SA}
+import tsec.common.SecureRandomId
+import tsec.jws.mac.JWTMac
+import tsec.mac.imports.HMACSHA256
+
+import cats.effect.IO
+import cats.implicits._
+
+class TSecAuthServiceSpec extends AuthenticatorSpec {
+
+  behavior of "Composability of TSecAuthService"
+
+    it should "compile when combined via SemigroupK" in {
```

Comment:
Should compile when combined via semigroupK isn't exactly the condition we want. That's what this PR is about, but not what the test itself should reflect.

"Should apply left" or apply right ish tests to just confirm the behavior is working fine for semigroupK is what we want, because it also gives us a guarantee that somewhere up the pipeline we didnt do something funky

Reply:
For an example, see: https://github.com/http4s/http4s/blob/215a84334b0a7b116f5b23f5feaf446843bf5054/server/src/test/scala/org/http4s/server/middleware/authentication/AuthMiddlewareSpec.scala#L149

Reply:
Agreed. I’ll get a new commit  up in the next hour or so.